### PR TITLE
solution

### PR DESCRIPTION
**Pull request by AndriiYelieva**

### CODE REVIEW

File: src/arrayMethodSort.js [hunk 1]
```diff
@@ -5,7 +5,22 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (
+          compareFunction
+            ? compareFunction(this[i], this[j]) > 0
+            : String(this[i]) > String(this[j])
```
Comment (Viktor-Kost) [inline src/arrayMethodSort.js:13]: It's better to create a default comparing function outside applyCustomSort function and set it as a default value

Example: 
```
const compareAsString = (a, b) => {
...
}

...

[].__proto__.sort2 = function(compareFunction = compareAsString) {
...
```

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

File: src/arrayMethodSort.js [hunk 2]
```diff
@@ -1,11 +1,38 @@
...
+};
+
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = compareAsStrings) {
+    let count;
+
+    do {
+      count = 0;
```
Comment (etojeDenys) [inline src/arrayMethodSort.js:21]: we can use a boolean variable here

File: src/arrayMethodSort.js [hunk 3]
```diff
@@ -1,11 +1,38 @@
...
+
+      for (let i = 1; i < this.length; i++) {
+        const prev = this[i - 1];
+        const current = this[i];
+
+        if (compareFunction(prev, current) > 0) {
+          count++;
```
Comment (etojeDenys) [inline src/arrayMethodSort.js:28]: set here to true

File: src/arrayMethodSort.js [hunk 4]
```diff
@@ -1,11 +1,38 @@
...
+          this[i - 1] = current;
+          this[i] = prev;
+        }
+      }
+    } while (count > 0);
```
Comment (etojeDenys) [inline src/arrayMethodSort.js:33]: and check if true here

**Review by etojeDenys — CHANGES_REQUESTED** [review]
it will improve readability

File: src/arrayMethodSort.js [hunk 5]
```diff
@@ -1,11 +1,24 @@
 'use strict';
 
-/**
- * Implement method Sort
- */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
     // write code here
+    for (let i = 0; i < this.length - 1; i++) {
```
Comment (IvanFesenko) [inline src/arrayMethodSort.js:6]: Create separate `sortString` that will be used if no `compareFunction` 

```suggestion
  [].__proto__.sort2 = function(compareFunction  = sortString) {
    // write code here
    for (let i = 0; i < this.length - 1; i++) {
```

File: src/arrayMethodSort.js [hunk 6]
```diff
@@ -1,11 +1,24 @@
...
+      for (let j = 0; j < this.length - i - 1; j++) {
+        const shouldSwap = compareFunction
+          ? compareFunction(this[j], this[j + 1]) > 0
+          : String(this[j]) > String(this[j + 1]);
```
Comment (IvanFesenko) [inline src/arrayMethodSort.js:10]:
```suggestion
        const shouldSwap = compareFunction(this[j], this[j + 1]) > 0;
```

**Review by IvanFesenko — CHANGES_REQUESTED** [review]

**Pull request closed without merging**